tests/test_storages.py

```diff
@@ -71,6 +73,29 @@ def test_json_readwrite(tmpdir):
     assert get('A short one') is None
 
 
+def test_create_dirs():
+    temp_dir = tempfile.gettempdir()
+    db_dir = ''
+    db_file = ''
+
+    while True:
```

Comment:
It probably will not hang. If you however used something like `getrandbytes` it may cause issues with valid path characters and such. Also you don't have to include the clean up though, I believe tmpdirs are [cleaned up automatically anyways](http://doc.pytest.org/en/latest/tmpdir.html#the-default-base-temporary-directory).